### PR TITLE
Add PVC tests and move XMX8 tests to a new folder

### PR DESCRIPTION
**Pull request by dkhaldi**

### CODE REVIEW

**Comment by dkhaldi** [comment]
@yubingex007-a11y, @myler please review

**Comment by dkhaldi** [comment]
failed test SYCL :: KernelAndProgram/multiple-kernel-linking.cpp unrelated to this change

**Review by myler — APPROVED** [review]

**Comment by myler** [comment]
The `matrix-xmx8` feature config looks good to me.

File: SYCL/Matrix/element_wise_irreg_sum_rows.cpp [hunk 1]
```diff
@@ -89,14 +89,14 @@ void matrix_sum_rows(queue q, big_matrix<T, M, N> &B, nd_range<2> &r) {
              for (int i = 0; i < data.length() / (TK / 4); i++) { // 4 per row
                // i*SG_SIZE index is found based on the round robin
                // distribution we are using in the implementation
-               sum_local_rows[row + global_idx * (TK / 4)] += data[i + row * 4]
+               sum_local_rows[row + global_idx * (TK / 4)] += data[i + row * 4];
```
Comment (dkhaldi) [inline SYCL/Matrix/element_wise_irreg_sum_rows.cpp]: @myler, this code was missing the semi colon but we were not getting the analysis that this test is failing at compile time.
How can we improve sharing the analysis of failures on the matrix code?

**Review by yubingex007-a11y — APPROVED** [review]

**Comment by dkhaldi** [comment]
@againull,  SYCL :: KernelAndProgram/kernel-bundle-merge-options-env.cpp is currently failing and unrelated to the changes I am making here.
Can you help approve/merge?

**Comment by dkhaldi** [comment]
@intel/llvm-reviewers-runtime, can you please help merge this?
 SYCL :: KernelAndProgram/kernel-bundle-merge-options-env.cpp is currently failing and unrelated to the changes I am making here.

**Comment by steffenlarsen** [comment]
Am I reading the tests correctly that the only differences between the existing tests and their XMX variants are the `SG_SIZE` and LIT feature requirements? If so, it may be easier to maintain the tests if the test implementations are instead moved to common header files that do not define the `SG_SIZE` macro.

For example for SYCL/Matrix/XMX8/element_wise_all_ops_bf16.cpp and SYCL/Matrix/element_wise_all_ops_bf16.cpp you would add a common header element_wise_all_ops_bf16_impl.hpp with the test implementation without `SG_SIZE` and then change the .cpp files to something like

```c++
//==----------- element_wise_all_ops_bf16.cpp  - DPC++ joint_matrix---------==//
//
// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
// See https://llvm.org/LICENSE.txt for license information.
// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
//
//===----------------------------------------------------------------------===//
// REQUIRES: matrix
// RUN: %clangxx -fsycl %s -o %t.out
// RUN: %CPU_RUN_PLACEHOLDER %t.out
// RUN: %GPU_RUN_PLACEHOLDER %t.out

#define SG_SIZE 16

#include "element_wise_all_ops_bf16_impl.hpp"
```

**Comment by dkhaldi** [comment]
> Am I reading the tests correctly that the only differences between the existing tests and their XMX variants are the `SG_SIZE` and LIT feature requirements? If so, it may be easier to maintain the tests if the test implementations are instead moved to common header files that do not define the `SG_SIZE` macro.

This is a very good suggestion. I just made the change.
@steffenlarsen, can you please take a look?

**Review by steffenlarsen — APPROVED** [review]
LGTM! 😄

**Comment by againull** [comment]
@dkhaldi Could tyu please fix clang-format failure.

**Comment by dkhaldi** [comment]
> @dkhaldi Could tyu please fix clang-format failure.

@againull done

**Comment by steffenlarsen** [comment]
Failures are unrelated and are most likely due to out-of-sync build being used with newer tests. Merging this.

**Pull request merged**